File: web/src/main/java/by/sorface/passport/web/facade/clients/ApplicationClientFacade.java
package by.sorface.passport.web.facade.clients;

import by.sorface.passport.web.records.requests.ApplicationClientPatchRequest;
import by.sorface.passport.web.records.requests.ApplicationRegistry;
import by.sorface.passport.web.records.responses.ApplicationClient;
import by.sorface.passport.web.records.responses.ApplicationClientRefreshSecret;

import java.util.List;
import java.util.UUID;

public interface ApplicationClientFacade {

    List<ApplicationClient> findByCurrentUser();

    ApplicationClient getByIdAndCurrentUser(final UUID clientId);

    ApplicationClient registry(final ApplicationRegistry registryClient);

    ApplicationClient partialUpdate(final UUID clientId, final ApplicationClientPatchRequest request);

    ApplicationClientRefreshSecret refreshSecret(final UUID clientId);

    void delete(final UUID clientId);

}

File: web/src/main/kotlin/by/sorface/passport/web/facade/clients/ApplicationClientFacade.kt
package by.sorface.passport.web.facade.clients

import by.sorface.passport.web.records.requests.ApplicationClientPatchRequest
import by.sorface.passport.web.records.requests.ApplicationRegistry
import by.sorface.passport.web.records.responses.ApplicationClient
import by.sorface.passport.web.records.responses.ApplicationClientRefreshSecret
import java.util.*

interface ApplicationClientFacade {
    fun findByCurrentUser(): List<ApplicationClient>

    fun getByIdAndCurrentUser(clientId: UUID): ApplicationClient?

    fun registry(registryClient: ApplicationRegistry): ApplicationClient?

    fun partialUpdate(clientId: UUID, request: ApplicationClientPatchRequest): ApplicationClient?

    fun refreshSecret(clientId: UUID): ApplicationClientRefreshSecret?

    fun delete(clientId: UUID)
}
